### ISSUE TITLE
SSD_mobilenet_v1/0.75_quantized_coco trained model is not detecting anything after porting on Android [Detect app]

### ISSUE THREAD

**Issue by alokranjan007** [tensorflow/tensorflow#21839]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
     No, I have only changed few lines in the config file.
- **OS Platform and Distribution: 14.04 LTS
- **Mobile device if the issue happens on mobile device**: Motoroloa, Android version 7.1.1
- **TensorFlow installed from (source or binary)**: Source
- **TensorFlow version (use command below)**: 1.10.0
- **Python version**: 3.6
- **GCC/Compiler version (if compiling from source)**: 7.2.0
- **CUDA/cuDNN version**: NA
- **GPU model and memory**: 4GB, 64-Bit
- **Bazel Version**: 0.16.0

### Describe the problem
Could you please help me regarding the results of the trained model.

I trained two different models from tensorflow zoo directory named ssd_mobilenet_v1_0.75_depth_quantized_coco and ssd_mobilenet_v1_quantized_coco each with 10000 training steps. I had changed the no of classes to 5 (for my data set) and used a batch size of 10 for both the models. However, the total no of of images are limited to 205.

Both the model trained well with all files generated. I **tested** the trained model using **webcam** and it worked perfectly fine.

As a next step, I used this blog by @achowdhery  [Blog Link](https://medium.com/tensorflow/training-and-serving-a-realtime-mobile-object-detector-in-30-minutes-with-cloud-tpus-b78971cf1193) here and converted the .pb file into .tflite format. The generated detect.tflite is of size 3.2 Mb (This is what authors have also suggested in the blog that the file should be less than 4 Mb). However, after build over android device, **there is no detection at all**.

I tried to hold it for at least 2 minutes for each classes (by showing images of each class), but there is no result in detection. It did not even overfit and show any false detection.

What I have done wrong? Please share your thoughts.

I am trying with both the models but no luck at all. I wonder the same model **(frozen graph) is working perfectly fine for webcam based detection.**

Thank you,

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
Exact command to reproduce

**Comment by alokranjan007** [comment]
@tensorflowbutler Thank you I have now updated.

**Comment by alokranjan007** [comment]
Is there any update on the issue?

**Comment by achowdhery** [comment]
@alokranjan007 Please share the intermediate files: checkpoint, frozen graph, pipeline config, label map.txt to reproduce the error. Is this quantized or float model?

**Comment by alokranjan007** [comment]
@achowdhery Thank you for your concern. I am unable to upload here all files due to limited upload permission. Sorry for the inconvenience, I am attaching here 2 different folders. Each following folder contains below files.
1. Source_file1: detect.tflite, label_map.txt and pipeline.config
2. Source_file2: tflite_graph.pb

Checkpoint file for 10000 training steps is of file size 44.7 Mb, which I am unable to upload due to limited upload permissions from GitHub.

The model is quantized version and I have downloaded it from tensorflow zoo directory. The folder name is "ssd_mobilenet_v1_quantized_coco".


I had also tried with ssd_mobilenet_0.75_quantized_coco, but after training no detection on the detect app. However, when I am checking the trained model using webcam it is working fine for both the models.

Thank you for your time.






[source_files1.tar.gz](https://github.com/tensorflow/tensorflow/files/2326689/source_files1.tar.gz)
[source_file2.tar.gz](https://github.com/tensorflow/tensorflow/files/2326695/source_file2.tar.gz)

**Comment by cy89** [comment]
@achowdhery thanks for responding. Do you have what you need from @alokranjan007 ?

**Comment by alokranjan007** [comment]
@cy89 I believe the source files have the necessary information. But, if any thin is missing then please let me know @achowdhery. I would be happy to learn more.

Thank you both for your help.

**Comment by alokranjan007** [comment]
Hey @cy89, Is there any update on the issue? I am still facing the same issue for detection.

**Comment by cy89** [comment]
@achowdhery, gentle ping?

**Comment by alokranjan007** [comment]
Its been 14 days...is there any update on the issue?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @achowdhery, @cy89: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @achowdhery, @cy89: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @achowdhery, @cy89: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignees @achowdhery, @cy89: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by achowdhery** [comment]
@alokranjan007 Is this still an issue? The app is working correctly on my end with Tensorflow Lite 1.12. May I request you to check with that version?

**Comment by ozzcet** [comment]
please add coma between output layer members.

**Comment by MeghnaNatraj** [comment]
Marking issue as resolved due to inactivity. Feel free to re-open this if it's unresolved or file a [new issue](https://github.com/tensorflow/tensorflow/issues/new/choose)
